state-network.md

```diff
@@ -125,54 +86,107 @@ A node should track their own radius value and provide this value in all Ping or
 
 ### Data Types
 
+#### OFFER/ACCEPT vs FINDCONTENT/FOUNDCONTENT payloads
+
+The data payloads for many content types in the history network differ between OFFER/ACCEPT and FINDCONTENT/FOUNDCONTENT.
+
+The OFFER/ACCEPT payloads need to be provable by their recipients.  These proofs are useful during OFFER/ACCEPT because they verify that the offered data is indeed part of the canonical chain.
+
+The FINDCONTENT/FOUNDCONTENT payloads do not contain proofs because a piece of state can exist under many different state roots.  All payloads can still be proved to be the correct requested data, however, it is the responsibility of the requesting party to anchor the returned data as canonical chain state data.
+
+
 #### Component Data Elements
 
-#### Proofs
+#### Merkle Patricia Trie (MPT) Proofs
+
 Merkle Patricia Trie (MPT) proofs consist of a list of witness nodes that correspond to each trie node that consists of various data elements depending on the type of node (e.g.blank, branch, extension, leaf).  When serialized, each witness node is represented as an RLP serialized list of the component elements with the largest possible node type being the branch node which when serialized is a list of up to sixteen hashes in `Bytes32` (representing the hashes of each of the 16 nodes in that branch and level of the tree) plus the 4 elements of the node's value (balance, nonce, codehash, storageroot) represented as `Bytes32`.  When combined with the RLP prefixes, this yields a possible maximum length of 667 bytes.  We specify 1024 as the maximum length due to constraints in the SSZ spec for list lengths being a power of 2 (for easier merkleization.)
+
+> TODO: Define proof such that there is one canonical representation of a witness that is streamable and minimal??
+> Specs should specify order of proof nodes, and disallow inclusion of superflous nodes.
+
 ```
 WitnessNode            := ByteList(1024)
 MPTWitness             := List(witness: WitnessNode, max_length=32)
 ```
 
-#### Account Trie Proof
+#### Paths (Nibbles)
 
-A leaf node from the main account trie and accompanying merkle proof against a recent `Header.state_root`
+A naive approach to storage of trie nodes would be to simply use the `node_hash` value of the trie node for storage.  This scheme however results in stored data not being tied in any direct way to it's location in the trie.  In a situation where a participant in the DHT wished to re-gossip data that they have stored, they would need to reconstruct a valid trie proof for that data in order to construct the appropriate OFFER/ACCEPT payload.  We include the `path` metadata in state network content keys so that it is possible to reconstruct this proof.
```

Comment:
If I understand correctly, we store path (nibbles) only in order to recreate proof for the purpose of re-gossiping, right? In that case, we would have to store either block number or state root we are anchored to.

Alternative is to just store hash of a parent node we are anchored to.

Reply:
Hash of parent node still anchors you to a specific state root.  Storing nibbles path allows you to reconstuct from *any* state root

Reply:
You are correct that storing path is intended to allow re-construction of proof.  Primary purpose is to be able to re-gossip (POKE).  You are also correct that we would need to store block number/hash in order to reliably reproduce the proof.

I think the problem that could/would arise if we chose to instead only store parent hash would be if the parent node was missing from the network.  This would be a largely un-recoverable position since any breakage in the parent chain would leave the node un-anchored.

By storing parent hash and nibbles, we keep enough data to do things like check the next block, or even reach out to external data sources to reconstruct the proof data in order to re-seed the missing parent node as well.  It does take up a few more bytes though.